cime_config/machines/config_machines.xml

```diff
@@ -4251,26 +4251,27 @@ commented out until "*** No rule to make target '.../libadios2pio-nm-lib.a'" iss
         <command name="unload">cray-parallel-netcdf</command>
         <command name="unload">cray-netcdf</command>
         <command name="unload">cray-hdf5</command>
-        <command name="unload">PrgEnv-gnu</command>
-        <command name="unload">PrgEnv-intel</command>
-        <command name="unload">PrgEnv-nvidia</command>
-        <command name="unload">PrgEnv-cray</command>
-        <command name="unload">PrgEnv-aocc</command>
         <command name="unload">intel</command>
         <command name="unload">intel-oneapi</command>
         <command name="unload">nvidia</command>
         <command name="unload">aocc</command>
         <command name="unload">cudatoolkit</command>
         <command name="unload">climate-utils</command>
+        <command name="unload">cray-libsci</command>
         <command name="unload">craype-accel-nvidia80</command>
         <command name="unload">craype-accel-host</command>
         <command name="unload">perftools-base</command>
         <command name="unload">perftools</command>
         <command name="unload">darshan</command>
+        <command name="unload">PrgEnv-gnu</command>
+        <command name="unload">PrgEnv-intel</command>
+        <command name="unload">PrgEnv-nvidia</command>
+        <command name="unload">PrgEnv-cray</command>
+        <command name="unload">PrgEnv-aocc</command>
```

Comment:
I found that these needed to be unloaded after their corresponding compiler modules or there would be an error about an undefined environment variable name.